### ISSUE TITLE
`object-curly-newline`: `minProperties` option doesn't respect comments

### ISSUE THREAD

**Issue by EDM115** [eslint-stylistic/eslint-stylistic#971]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Here's a simple code example :
```vue
<!-- UiList.vue -->
<template>
  <ul :class="classes">
    <slot />
  </ul>
</template>

<script setup lang="ts">
const props = defineProps<{
  /**
   * Disables the minimum width
   */
  compact?: boolean;
}>()

const classes = computed(() => [
  "ui-list",
  props.compact
    ? "ui-list--compact"
    : "",
])
</script>

<style scoped lang="scss">
.ui-list {
  list-style: none;
  margin: 0;
  padding: .25rem;
  min-width: 160px;

  &--compact {
    min-width: 0px;
  }
}
</style>
```

in this situation, the "comment" above the compact prop serves as JSDoc so when you hover over it in VS Code for example you get a description of what it does.  
but when you enable `@stylistic/object-curly-newline` with `minProperties: 2` (or more), here's what you get :
```
UiList.vue
   8:27  error  Unexpected line break after this opening brace   @stylistic/object-curly-newline
  13:1   error  Unexpected line break before this closing brace  @stylistic/object-curly-newline
```

The "fix" becomes :
```diff
  const props = defineProps<{
    /**
     * Disables the minimum width
     */
-   compact?: boolean;
- }>()
+ compact?: boolean;}>()
```
And the first line stays in error without an available fix.  
This rule should consider comments (or at least JSDoc ones) as an "element" that counts towards `minProperties`

cc #675 (thanks for the concise title)

### Reproduction

https://stackblitz.com/edit/github-getdv6l3?file=src%2FUiList.vue

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
> This rule should consider comments (or at least JSDoc ones) as an "element" that counts towards `minProperties`

I think it is not a good solution. As noticed in https://github.com/eslint-stylistic/eslint-stylistic/issues/675#issuecomment-2629523695:
> Counting comments towards the "elements" would break the rule for those who want single-comment blocks to be on a single line and single-statement blocks on multiple lines.

The best solution I can think of now is to ignore `minProperties` when there is a comment in the node. Because when there are comments in the node, it is difficult to maintain a single-line display.

**Comment by EDM115** [comment]
this seems more logical, I agree

**Comment by 9romise** [comment]
After taking a look, I think setting `consistent` to `true` might resolve the issue. Would you like to give it a try?

**Comment by EDM115** [comment]
@9romise thanks, it indeed fixes it !
